src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -0,0 +1,535 @@
+                builder.Tags.AddRange(Tags);
+            }
+
+            builder.Description = Description;
+            builder.Copyright = Copyright;
+            builder.ReleaseNotes = ReleaseNotes;
+            Uri tempUri;
+            if (Uri.TryCreate(LicenseUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.LicenseUrl = tempUri;
+            }
+            if (Uri.TryCreate(ProjectUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.ProjectUrl = tempUri;
+            }
+            if (Uri.TryCreate(IconUrl, UriKind.Absolute, out tempUri))
+            {
+                builder.IconUrl = tempUri;
+            }
+            builder.RequireLicenseAcceptance = RequireLicenseAcceptance;
+            if (!string.IsNullOrEmpty(RepositoryUrl) || !string.IsNullOrEmpty(RepositoryType))
+            {
+                builder.Repository = new RepositoryMetadata(RepositoryType, RepositoryUrl);
+            }
+            builder.PackageTypes = ParsePackageTypes();
+            ParseProjectToProjectReferences(builder, packArgs);
+            GetPackageReferences(builder);
+            return builder;
+        }
+
+        private ICollection<PackageType> ParsePackageTypes()
+        {
+            var listOfPackageTypes = new List<PackageType>();
+            if (PackageTypes != null)
+            {
+                foreach (var packageType in PackageTypes)
+                {
+                    string[] packageTypeSplitInPart = packageType.Split(new char[] {','});
+                    string packageTypeName = packageTypeSplitInPart[0];
+                    var version = PackageType.EmptyVersion;
+                    if (packageTypeSplitInPart.Length > 1)
+                    {
+                        string versionString = packageTypeSplitInPart[1];
+                        Version.TryParse(versionString, out version);
+                    }
+                    listOfPackageTypes.Add(new PackageType(packageTypeName, version));
+                }
+            }
+            return listOfPackageTypes;
+        }
+
+        private LibraryDependency GetLibraryDependency(ITaskItem p2pReference, string packageId, string version)
+        {
+            LibraryIncludeFlags includeFlags, privateAssetsFlag;
+            GetAssetMetadata(p2pReference, out includeFlags, out privateAssetsFlag);
+
+            LibraryRange libraryRange = new LibraryRange(packageId, VersionRange.Parse(version, true), LibraryDependencyTarget.All);
+            var libDependency = new LibraryDependency()
+            {
+                LibraryRange = libraryRange,
+                IncludeType = includeFlags,
+                SuppressParent = privateAssetsFlag
+            };
+            return libDependency;
+        }
+
+        private void InitCurrentDirectoryAndFileName(PackArgs packArgs)
+        {
+            if (PackItem == null)
+            {
+                throw new InvalidOperationException(nameof(PackItem));
```

Comment:
`ArgumentException` and/or a more descriptive error message would be more useful than:

    System.InvalidOperationException: PackItem